main.go

```diff
@@ -215,6 +347,18 @@ func main() {
 		glog.Fatalf("Failed to create client: %v", err)
 	}
 
+	supportedResourceMetaInfo, err := getServerSupportedResourceMetaInfo(kubeClient)
+	if err != nil {
+		glog.Fatalf("Failed to list server supported resource meta info: %v", err)
+	}
+
+	glog.V(4).Info(supportedResourceMetaInfo)
+
+	err = validateCollectorsAgainstServer(supportedResourceMetaInfo, collectors)
```

Comment:
while I agree this is a good behavior, it's not backward compatible, I think we need to hide this behind a flag

Reply:
> while I agree this is a good behavior, it's not backward compatible, 

IMHO, this should be done earlier and it is a prerequisite condition before kube-state-metrics start against any Kubernetes server. Fatal things should fail fast instead of just logging the error like what we do for now. 

There are two conditions that the check may not pass:
* the collector is not available
* the currently used collector group version is not available

For the first one, we already have `--collectors` command line argument to specify a collectors whitelist. For the second one, in this PR we have the ability to customize the collector group version for now. So, if the default collector group version does not supported against the server, we can use `--collectors-config` to customize the group version for the collector.

So, i think this may not be backward incompatible. :) 

@brancz WDYT.

Reply:
hmm ... shouldn't we just disable collectors that are not available and log that?

Reply:
Seems fine to me. There are two ways to inform end users about this error. Log and ignore or fail fast. I am in favor of fail fast. But log and ignore is also good to me. :)

Done. PTAL.